src/conan-utils.stanza

```diff
@@ -0,0 +1,360 @@
+<DOC>
+defn conan-get-package-ids-for-revision (package_name:String, package_version:String, recipe_revision:String,
+                                         curl:Curl, -- headers:Tuple<String> = [], repourl:String = DEFAULT-CONAN-URL) -> Tuple<KeyValue<String, JSON>>:
+  ; search for available package_ids of the recipe revision
+  val queryurl = to-string("%_/v2/conans/%_/%_/_/_/revisions/%_/search" % [repourl, url-encode(package_name), url-encode(package_version), url-encode(recipe_revision)])
+  val response = read-url(curl, headers, queryurl)
+  val jresult = parse(response) as JObject
+  if key?(jresult, "errors"):
+    throw(Exception("Conan error while getting package_ids for recipe revision \"%_/%_#%_\": \"%_\"" % [package_name, package_version, recipe_revision, to-json-string(jresult)]))
+  entries(jresult)
+
+
+defn conan-get-package-revisions (package_name:String, package_version:String, recipe_revision:String, package_id:String,
+                                  curl:Curl, -- headers:Tuple<String> = [], repourl:String = DEFAULT-CONAN-URL) -> Tuple<JSON>:
+  ; search for available package revisions of the given recipe revision and package_id
+  val queryurl = to-string("%_/v2/conans/%_/%_/_/_/revisions/%_/packages/%_/revisions" % [
+                           repourl, url-encode(package_name), url-encode(package_version), url-encode(recipe_revision), url-encode(package_id)])
+  val response = read-url(curl, headers, queryurl)
+  val jresult = parse(response) as JObject
+  if key?(jresult, "errors"):
+    throw(Exception("Conan error while getting package revisions for package_id \"%_/%_#%_\": \"%_\"" % [package_name, package_version, recipe_revision, to-json-string(jresult)]))
+  jresult["revisions"] as Tuple<JSON>
+
+
+doc: \<DOC>
+Searches the given conan repository for the latest package matching the given ConanVersion and options.
+@throws Exception on failure or package not found
+<DOC>
+public defn conan-fully-qualify-latest-version (cv:ConanVersion, -- options:TomlTable = TomlTable(), repourl:String = DEFAULT-CONAN-URL) -> ConanVersion :
+  label<ConanVersion> return :
+    debug("conan-fully-qualify-latest-version: qualifying version: %_" % [to-string(cv)])
+    ; If the cv has already has all of the parts, then return it unchanged
+    if not (package_id(cv) == false or recipe_revision(cv) == false or package_revision(cv) == false):
+      return(cv)
+    else:
+      ; cv was missing a revision or package_id, so search the server for the latest matching package
+      ; curl $APIBASE/conans/pcre/8.45/_/_/search
+      val headers = ["Content-Type: application/json"]
+      var curl
+      try :
+        curl = Curl()
+
+        val package_name = name(cv)
+        val package_version = version(cv)
+
+        val current_conan_os = match(CURRENT-PLATFORM) :
+          (p:Linux) : "Linux"
+          (p:OS-X) : "Macos"
+          (p:Windows) : "Windows"
+        val current_conan_os_lc = lower-case(current_conan_os)
+
```

Comment:
You might want to make this a separate function and make it clear that we are converting into strings that match `Conan`'s expectation of what the OS name is.

At first I was going to recommend this: 

```
lower-case $ to-string(CURRENT-PLATFORM)
```

But it seems like the issue is that conan expects `Macos` instead of `os-x`

Reply:
Done